packages/components/src/core/NavigationHeader/components/NavigationHeaderPrimaryNav/index.tsx

```diff
@@ -0,0 +1,49 @@
+import { SdsTagColorType } from "src/core/Tag";
+import { PrimaryNavItem, StyledLabel } from "./style";
+import { ReactNode } from "react";
+import { StyledTag } from "../../style";
+import { StyledSection } from "../style";
+
+export interface NavigationHeaderPrimaryNavItem<T extends string> {
+  key: T;
+  label: ReactNode;
+  tag?: string;
+  tagColor?: SdsTagColorType;
+}
+
+export interface NavigationHeaderPrimaryNavProps<T extends string> {
+  hasInvertedStyle?: boolean;
+  items: NavigationHeaderPrimaryNavItem<T>[];
+  onChange(value: T): void;
+  value: T;
+}
+
+export default function NavigationHeaderPrimaryNav<T extends string>({
+  hasInvertedStyle,
+  items,
+  onChange,
+  value,
+}: NavigationHeaderPrimaryNavProps<T>) {
+  return (
+    <StyledSection>
+      {items.map((item) => {
+        const isActive = item.key === value;
+
+        return (
+          <PrimaryNavItem
```

Comment:
not a blocker for this PR, but is there a way to accept ReactNode or somehow allow rendering Next/Link and/or `<a />` for links instead of onClick driven navigation? Just thinking that would help with semantic HTML + a11y!

Reply:
In other components we export the subcomponents to allow the users to compose to maintain the flexibility and retain the styling consistency. In this case, perhaps exporting `PrimaryNavItem` could help? That way the Nav Header can accept ReactNode and the user can wrap `PrimaryNavItem` with a Link component if needed? 

Something like that, probably still need to think through together 😂

Reply:
@codemonkey800, I agree with having a `LinkComponent`, it makes total sense. I’ve also been thinking about refactoring some of our other components to use a polymorphic architecture for better control over the underlying tags. 😬🥸